overlord/ifacestate/apparmorprompting/common/common.go

```diff
@@ -68,7 +69,7 @@ const (
 // If the given permission is not found in the list, returns an error, along
 // with the original list.
 func RemovePermissionFromList(list []PermissionType, permission PermissionType) ([]PermissionType, error) {
-	newList := make([]PermissionType, 0, len(list)-1)
+	newList := make([]PermissionType, 0, len(list))
```

Comment:
If we allocate space for `len(list)-1`, but `len(list) == 0`, then we get a panic. Since this function is useful for efficiently and idempotently removing a permission if it exists (else doing nothing). However, allocating `len(list)-1` is desirable in all cases except where `len(list) == 0`. Thus, I will add a check before this line that returns early if `len(list) == 0`, and otherwise allocates `len(list)-1`.